python/spacewalk/satellite_tools/spacewalk-data-fsck

```diff
@@ -357,6 +359,27 @@ def is_srpm(file):
     else:
         return False
 
+def is_orphaned_rpm(package_id: int) -> bool:
+    if not isinstance(package_id, int):
+        # Prevent possible SQL injections
+        raise ValueError("package_id must be an integer")
+
+    query = f"""
+    SELECT EXISTS (
+      SELECT 1 FROM rhnPackage p
+      JOIN rhnChannelPackage c ON c.package_id = p.id WHERE p.id = {package_id}
+    )
+    """
+
+    h = rhnSQL.prepare(query)
+    h.execute()
+    row = h.fetchone_dict()
+    if not row:
+        log(1, "RPM not in DB")
+        return False
```

Comment:
I think "not row" can not happen as "select exists( ... )  would always return a row either true or false.
Maybe the log message should be removed and replaces with a comment?

Reply:
True, good catch!